tests/rancher-istio/rancher_istio.go

```diff
@@ -0,0 +1,849 @@
+				),
+		},
+
+		{
+			Name: "Set .Values.istiodRemote to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"istiodRemote.enabled", true,
+				),
+		},
+
+		{
+			Name: "Set .Values.istiodRemote to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"istiodRemote.enabled", false,
+				),
+		},
+
+		{
+			Name: "Set .Values.nodeSelector",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"nodeSelector", defaultNodeSelector,
+				),
+		},
+
+		{
+			Name: "Set .Values.tolerations",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"tolerations", testTolerations,
+				),
+		},
+
+		{
+			Name: "Set .Values.installer repository and tag",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set("installer", map[string]string{
+					"repository":            "testRepo/mirrored-istio-installer",
+					"tag":                   "1.7.3",
+					"releaseMirror.enabled": "false",
+					"debug.secondsSleep":    "0",
+				}),
+		},
+
+		{
+			Name: "Set CNI to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.enabled", "false",
+				),
+		},
+
+		{
+			Name: "Set CNI to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set("cni", map[string]interface{}{
+					"enabled":    true,
+					"repository": "testRepo/mirrored-istio-install-cni",
+					"tag":        "1.7.3",
+					"logLevel":   "debug",
+					"excludeNamespaces": []string{
+						"kube-system",
+						"istio-system",
+					},
+				}),
+		},
+
+		{
+			Name: "Set .Values.pilot.enabled to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"pilot.enabled", false,
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.enabled to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set("pilot", map[string]interface{}{
+					"enabled":             true,
+					"repository":          "testRepo/mirrored-istio-pilot",
+					"tag":                 "1.7.3",
+					"hpaSpec":             testHPASpec,
+					"podDisruptionBudget": testPodDisruptionBudget,
+				}),
+		},
+
+		{
+			Name: "Set .Values.egressGateways",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set("egressGateways", map[string]interface{}{
+					"enabled":             false,
+					"type":                "NodePort",
+					"hpaSpec":             testHPASpec,
+					"podDisruptionBudget": testPodDisruptionBudget,
+				}),
+		},
+
+		{
+			Name: "Set .Values.ingressGateways",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set("ingressGateways", map[string]interface{}{
+					"enabled":             true,
```

Comment:
Shouldn't there be a test for ingressGateway set to false?

Reply:
Added